src/main/java/core/basesyntax/utils/transaction/handlers/OperationHandlerFactory.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.utils.transaction.handlers;
+
+import core.basesyntax.utils.transaction.FruitTransaction;
+import core.basesyntax.utils.transaction.handlers.impl.BalanceOperation;
+import core.basesyntax.utils.transaction.handlers.impl.PurchaseOperation;
+import core.basesyntax.utils.transaction.handlers.impl.ReturnOperation;
+import core.basesyntax.utils.transaction.handlers.impl.SupplyOperation;
+
+public class OperationHandlerFactory {
+    public OperationHandler getOperationHandlerFromOperation(FruitTransaction.Operation operation) {
+        return switch (operation) {
+            case BALANCE -> new BalanceOperation();
+            case PURCHASE -> new PurchaseOperation();
+            case RETURN -> new ReturnOperation();
+            case SUPPLY -> new SupplyOperation();
+        };
```

Comment:
Avoid using switch-cases for operation handling. It is recommended to use the Strategy pattern instead. Create a map of strategies and retrieve the correct strategy based on the operation. This will make it easier to extend the functionality in the future.